pad.go

```diff
@@ -80,20 +81,23 @@ func (pad *PAD) Update(policies Policies) {
 }
 
 func (pad *PAD) Set(key string, value []byte) error {
-	return pad.tree.Set(key, value)
+	index := pad.computePrivateIndex(key)
+	return pad.tree.Set(index, key, value)
 }
 
 func (pad *PAD) Lookup(key string) *AuthenticationPath {
 	str := pad.currentSTR
-	return str.tree.Get(key)
+	lookupIndex := pad.computePrivateIndex(key)
+	return str.tree.Get(lookupIndex)
 }
 
 func (pad *PAD) LookupInEpoch(key string, epoch uint64) (*AuthenticationPath, error) {
 	str := pad.GetSTR(epoch)
 	if str == nil {
 		return nil, ErrorSTRNotFound
 	}
-	ap := str.tree.Get(key)
+	lookupIndex := pad.computePrivateIndex(key)
```

Comment:
> And I'm assuming that the private key change in the policy means we need to recompute all the indices.

Yes. This would be a good feature to expose to the developer as this also has the nice side-effect of re-randomizing the order of leaves in the tree (as discussed in https://github.com/coniks-sys/libmerkleprefixtree-go/pull/19#issuecomment-233067359).